cli/src/commands/sidekick.rs

```diff
@@ -0,0 +1,209 @@
+		let (api_endpoint, _token) = global_config::read_project(|x| {
+			(x.cluster.api_endpoint.clone(), x.tokens.cloud.clone())
+		})
+		.await?;
+
+		// Create OpenAPI configuration without bearer token to send link request
+		let openapi_config_cloud_unauthed = rivet_api::apis::configuration::Configuration {
+			base_path: api_endpoint
+				.clone()
+				.unwrap_or_else(|| ctx::DEFAULT_API_ENDPOINT.to_string()),
+			user_agent: Some(ctx::user_agent()),
+			..Default::default()
+		};
+
+		// Prepare the link
+		let prepare_res = rivet_api::apis::cloud_devices_links_api::cloud_devices_links_prepare(
+			&openapi_config_cloud_unauthed,
+		)
+		.await;
+		if let Err(err) = prepare_res.as_ref() {
+			struct_fmt::print(
+				&Format::Json,
+				&SideKickResponse::Err(json!({
+					"error": err.to_string(),
+				})),
+			)?;
+			bail!("Error: {err:?}");
+		}
+		let prepare_res = prepare_res.context("cloud_devices_links_prepare")?;
+
+		struct_fmt::print(
+			&Format::Json,
+			&SideKickResponse::Ok(json!({
+				"device_link_url": prepare_res.device_link_url,
+				"device_link_token": prepare_res.device_link_token,
+			})),
+		)?;
+
+		Ok(())
+	}
+
+	pub async fn wait_for_login(&self, token: &String) -> Result<()> {
+		let (api_endpoint, _token) = global_config::read_project(|x| {
+			(x.cluster.api_endpoint.clone(), x.tokens.cloud.clone())
+		})
+		.await?;
+
+		// Create OpenAPI configuration without bearer token to send link request
+		let openapi_config_cloud_unauthed = rivet_api::apis::configuration::Configuration {
+			base_path: api_endpoint
+				.clone()
+				.unwrap_or_else(|| ctx::DEFAULT_API_ENDPOINT.to_string()),
+			user_agent: Some(ctx::user_agent()),
+			..Default::default()
+		};
+
+		// Wait for link to complete
+		let mut watch_index = None;
+		let token = loop {
+			let prepare_res = rivet_api::apis::cloud_devices_links_api::cloud_devices_links_get(
+				&openapi_config_cloud_unauthed,
+				&token,
+				watch_index.as_ref().map(String::as_str),
+			)
+			.await;
+			if let Err(err) = prepare_res.as_ref() {
+				struct_fmt::print(
+					&Format::Json,
+					&SideKickResponse::Err(json!({
+						"error": err.to_string(),
+					})),
+				)?;
+				bail!("Error: {err:?}");
+			}
+			let prepare_res = prepare_res.context("cloud_devices_links_get")?;
+
+			watch_index = Some(prepare_res.watch.index);
+
+			if let Some(token) = prepare_res.cloud_token {
+				break token;
+			}
+		};
+
+		// Create new context
+		let new_ctx = cli_core::ctx::init(
+			api_endpoint,
+			// Exclude overridden access token to check the token
+			token.clone(),
+		)
+		.await?;
+
+		// Inspect the token
+		let inspect_res =
+			rivet_api::apis::cloud_auth_api::cloud_auth_inspect(&new_ctx.openapi_config_cloud)
+				.await;
+		if let Err(err) = inspect_res.as_ref() {
+			struct_fmt::print(
+				&Format::Json,
+				&SideKickResponse::Err(json!({
+					"error": err.to_string(),
+				})),
+			)?;
+
+			bail!("Error: {err:?}");
```

Comment:
@NathanFlurry re: discussion on errors, I've changed each one to return a response early with the error so that it's easy to handle on the engine's side. This is pre-exploration on if we should match error handling to similar usage across the Rivet ecosystem, but we should chat tomorrow about if it's overengineering for the project's scope or not.